deepcell_spots/applications/polaris.py

```diff
@@ -23,227 +23,142 @@
-            image_mpp (float): Microns per pixel for ``image``.
-            pad_mode (str): The padding mode, one of "constant" or "reflect".
-            preprocess_kwargs (dict): Keyword arguments to pass to the
-                pre-processing function.
-            postprocess_kwargs (dict): Keyword arguments to pass to the
-                post-processing function.
-        Raises:
-            ValueError: Input data must match required rank, calculated as one
-                dimension more (batch dimension) than expected by the model.
-            ValueError: Input data must match required number of channels.
-        Returns:
-            numpy.array: Coordinate spot locations
-        """
-        # Check input size of image
-        if len(image.shape) != self.required_rank:
-            raise ValueError('Input data must have {} dimensions. '
-                             'Input data only has {} dimensions'.format(
-                                 self.required_rank, len(image.shape)))
-
-        if image.shape[-1] != self.required_channels:
-            raise ValueError('Input data must have {} channels. '
-                             'Input data only has {} channels'.format(
-                                 self.required_channels, image.shape[-1]))
-
-        # Resize image, returns unmodified if appropriate
-        resized_image = self._resize_input(image, image_mpp)
+    def __init__(self,
+                 segmentation_model=None,
+                 segmentation_compartment='cytoplasm',
+                 spots_model=None):
 
-        # Generate model outputs
-        output_images = self._run_model(
-            image=resized_image, batch_size=batch_size,
-            pad_mode=pad_mode, preprocess_kwargs=preprocess_kwargs
-        )
+        self.spots_app = SpotDetection(model=spots_model)
 
-        # Resize output_images back to original resolution if necessary
-        label_image = self._resize_output(output_images, image.shape)
+        valid_compartments = ['cytoplasm', 'nucleus', 'None']
+        if segmentation_compartment not in valid_compartments:
+            raise ValueError('Invalid compartment supplied: {}. '
+                             'Must be one of {}'.format(segmentation_compartment,
+                                                        valid_compartments))
 
-        # Postprocess predictions to create label image
-        predicted_spots = self._postprocess(label_image, **postprocess_kwargs)
-
-        return predicted_spots
+        if segmentation_compartment == 'cytoplasm':
+            self.segmentation_app = CytoplasmSegmentation(model=segmentation_model)
+            self.segmentation_app.preprocessing_fn = histogram_normalization
+            self.segmentation_app.postprocessing_fn = deep_watershed
+        elif segmentation_compartment == 'nucleus':
+            self.segmentation_app = NuclearSegmentation(model=segmentation_model)
+        else:
+            self.segmentation_app = None
```

Comment:
I think we should issue a warning here that a segmentation app will not be used and that spots cannot be mapped onto cells